azurerm/internal/services/iothub/validate/digital_twins.go

```diff
@@ -0,0 +1,15 @@
+package validate
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func DigitaltwinsName() schema.SchemaValidateFunc {
+	return validation.StringMatch(
+		regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9-]{1,61}[A-Za-z0-9]$`),
+		`Name contains invalid characters or exceeds allowed length.`,
+	)
+}
```

Comment:
Can we make this function more clear to the end user in case an error does occur? WDYT?
```suggestion
func digitalTwinsName(i interface{}, k string) (warnings []string, errors []error) {
	v, ok := i.(string)
	if !ok {
		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
		return
	}
	
	if len(v) < 3 {
		errors = append(errors, fmt.Errorf("length should be greater than %d, got %q", 3, v))
		return
	}
	
	if len(v) > 63 {
		errors = append(errors, fmt.Errorf("length should be less than %d, got %q", 63, v))
		return
	}
	
	if !regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9-]+[A-Za-z0-9]$`).MatchString(v) {
		errors = append(errors, fmt.Errorf("expected value of %s does not match regular expression, got %v", k, v))
		return
	}
	return
}
```